File: src/main/java/no/lambdaconsulting/javatokotlin/service/PetService.java
package no.lambdaconsulting.javatokotlin.service;

import no.lambdaconsulting.javatokotlin.dao.BreedDao;
import no.lambdaconsulting.javatokotlin.dao.PetDao;
import no.lambdaconsulting.javatokotlin.domain.PetDetails;
import org.springframework.stereotype.Service;

@Service
public class PetService {

    private final PetDao petDao;
    private final BreedDao breedDao;

    public PetService(PetDao petDao, BreedDao breedDao) {
        this.petDao = petDao;
        this.breedDao = breedDao;
    }

    public PetDetails getDetails(Integer id) {
        var petSummary = petDao.getById(id);
        var breed = breedDao.getById(petSummary.breedId);
        return new PetDetails(
                petSummary.id,
                petSummary.name,
                breed
        );
    }
}

File: src/main/kotlin/no/lambdaconsulting/javatokotlin/service/PetService.kt
package no.lambdaconsulting.javatokotlin.service

import no.lambdaconsulting.javatokotlin.dao.BreedDao
import no.lambdaconsulting.javatokotlin.dao.PetDao
import no.lambdaconsulting.javatokotlin.domain.PetDetails
import org.springframework.stereotype.Service

@Service
class PetService(private val petDao: PetDao, private val breedDao: BreedDao) {

    fun getDetails(id: Int) = petDao.getById(id)?.let { petSummary ->
        PetDetails(
            petSummary.id,
            petSummary.name,
            breedDao.getById(petSummary.breedId)
        )
    }
}